src/test/java/net/dv8tion/jda/test/modals/ModalTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.test.modals;
+
+import net.dv8tion.jda.api.components.Component;
+import net.dv8tion.jda.api.components.attribute.IDisableable;
+import net.dv8tion.jda.api.components.label.Label;
+import net.dv8tion.jda.api.components.label.LabelChildComponent;
+import net.dv8tion.jda.api.components.selections.EntitySelectMenu;
+import net.dv8tion.jda.api.components.selections.StringSelectMenu;
+import net.dv8tion.jda.api.modals.Modal;
+import net.dv8tion.jda.test.components.ComponentTestData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+public class ModalTest {
+    static Map<Component.Type, Class<? extends LabelChildComponent>> LABEL_CHILDREN = Map.of(
+            Component.Type.STRING_SELECT, StringSelectMenu.class,
+            Component.Type.USER_SELECT, EntitySelectMenu.class,
+            Component.Type.ROLE_SELECT, EntitySelectMenu.class,
+            Component.Type.MENTIONABLE_SELECT, EntitySelectMenu.class,
+            Component.Type.CHANNEL_SELECT, EntitySelectMenu.class);
+
+    @MethodSource("modalComponents")
+    @ParameterizedTest
+    public void testDisabledComponents(Component component) {
+        Assertions.assertInstanceOf(IDisableable.class, component);
```

Comment:
We use assertj in our tests:

```java
assertThat(component).isInstanceOf(IDisableable.class)
```

Reply:
I'll adjust all the assertions to AssertJ in that case.

Going to wait to push an update until we've gotten a path forward on the other aspects. Marking this as "resolved" since it's a straight forward project standard.